presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcess.java

```diff
@@ -148,6 +149,18 @@ public void close()
     {
         if (process != null && process.isAlive()) {
             process.destroy();
+            try {
+                process.waitFor(120, TimeUnit.SECONDS);
```

Comment:
nit: is it worth adding a log to indicate if the shutdown was graceful?

Reply:
Yup log is added in the finally{} block